### DISCUSSION TITLE
Fetch data from public folder

### DISCUSSION THREAD

**Discussion by janusznoszczynski** [nuxt/nuxt#16244] [Questions]
I have mydata.json file in public folder:
/public/data/mydata.json

In page there is a code:
const { data } = await useFetch(`/data/mydata.json`);

But I am getting an error when trying to fetch data from json public folder:
[Vue Router warn]: No match found for location with path "/data/mydata.json"
RefImpl {
  __v_isShallow: false,
  dep: undefined,
  __v_isRef: true,
  _rawValue: null,
  _value: null
}

When requesting it directly from browser I am getting correct json data in the response:
http://localhost:3000/data/my-data.json

**Comment by TheAlexLichter** [comment]
UPDATE: Created an issue, please follow https://github.com/nuxt/framework/issues/9427

---

Hey! Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

Also, see https://github.com/nuxt/framework/issues/7941. It *should* work fine.

**Comment by RamaHerbin** [comment]
Hi,
I've got the same behaviour.
I would like to relatively fetch a json from the public folder,
How can I do this without specifying `http://localhost:3000` ?
Am I forced to create a server API that `fs.read` the JSON files from the public folder ?

**Reply by TheAlexLichter** [reply]
Please read my comment from above ☺️
You don't have to. You can fetch via e.g. `useFetch('/my-image.png')`.

**Reply by RamaHerbin** [reply]
Thank you for your response.
I've read your comment but I still don't understand.
If I request the JSON file with `useFetch('/data.json')` , the request is null. I have to specify `useFetch('http://locahost:3000/data.json')` so it can work and I don't understand why.

Here is an [example](https://stackblitz.com/edit/nuxt-framework-jhsygu?file=pages/index.vue)

**Reply by TheAlexLichter** [reply]
You missed parentheses there. `data.value = (await useFetch('/data.json')).data; //null !`

[Improved StackBlitz](https://stackblitz.com/edit/nuxtframeworkdiscussions8358discussioncomment-4273658?file=pages%2Findex.vue)

**Reply by TheAlexLichter** [reply]
okay, nevermind. You are right!

**Reply by TheAlexLichter** [reply]
Issue is up @ https://github.com/nuxt/framework/issues/9427
